source/common/router/config_impl.h

```diff
@@ -922,6 +928,14 @@ class NullConfigImpl : public Config {
     return nullptr;
   }
 
+  RouteConstSharedPtr route(const RouteCallback& cb, const Http::RequestHeaderMap&,
+                            const StreamInfo::StreamInfo&, uint64_t) const override {
+    if (cb) {
+      cb(nullptr, RouteEvalStatus::NoMoreRoutes);
+    }
```

Comment:
I would just ignore cb here and return nullptr as we are not doing this elsewhere.